src/experiment_prototype/experiment_slice.py

```diff
@@ -289,6 +297,16 @@ class ExperimentSlice:
     pulse_sequence: conlist(conint(ge=0, strict=True), unique_items=True)
     rx_beam_order: list[Union[beam_order_type, conint(ge=0, strict=True)]]
 
+    # Frequency rx and tx limits are dependant on the tx and rx center frequencies. As such, since the center freq
+    # paramter is defined by slice, the max and min rx frequencies must be determined after ceter freq validation
+    txctrfreq: Optional[freq_khz] = None
+    rxctrfreq: Optional[freq_khz] = None
+    tx_minfreq: Optional[freq_float_hz] = options.min_freq
+    tx_maxfreq: Optional[freq_float_hz] = options.max_freq
+    rx_minfreq: Optional[freq_float_hz] = options.min_freq
+    rx_maxfreq: Optional[freq_float_hz] = options.max_freq
+
+
     # These fields have default values. Some have specification requirements in conjunction with each other
     # e.g. one of intt or intn must be specified.
     freq: Optional[freq_khz] = None
```

Comment:
This is a different thing, but I believe this should be required (what do we do if they don't specify a freq?)

Reply:
I was wondering about that. I think if freq is undefined it should raise a value error and fail out. I don't think it would make sense to have a default frequency.

Reply:
If freq is undefined and clrfreqsearch is also not set there is already a root validator that throws and error. So I've just removed the return nothing statement at the start of the freq validator